src/Core/Grid/Definition/Factory/CartRuleGridDefinitionFactory.php

```diff
@@ -0,0 +1,262 @@
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getColumns(): ColumnCollection
+    {
+        return (new ColumnCollection())
+            ->add((new BulkActionColumn('bulk'))
+            ->setOptions([
+                'bulk_field' => 'id_cart_rule',
+            ])
+            )
+            ->add((new DataColumn('id_cart_rule'))
+            ->setName($this->trans('ID', [], 'Admin.Global'))
+            ->setOptions([
+                'field' => 'id_cart_rule',
+            ])
+            )
+            ->add((new DataColumn('name'))
+            ->setName($this->trans('Name', [], 'Admin.Global'))
+            ->setOptions([
+                'field' => 'name',
+            ])
+            )
+            ->add((new DataColumn('priority'))
+            ->setName($this->trans('Priority', [], 'Admin.Global'))
+            ->setOptions([
+                'field' => 'priority',
+            ])
+            )
+            ->add((new DataColumn('code'))
+            ->setName($this->trans('Code', [], 'Admin.Global'))
+            ->setOptions([
+                'field' => 'code',
+            ])
+            )
+            ->add((new DataColumn('quantity'))
+            ->setName($this->trans('Quantity', [], 'Admin.Catalog.Feature'))
+            ->setOptions([
+                'field' => 'quantity',
+            ])
+            )
+            ->add((new DateTimeColumn('date_to'))
+            ->setName($this->trans('Expiration date', [], 'Admin.Catalog.Feature'))
+            ->setOptions([
+                'format' => 'Y-m-d H:i',
+                'field' => 'date_to',
+            ])
+            )
+            ->add((new ToggleColumn('active'))
+            ->setName($this->trans('Enabled', [], 'Admin.Global'))
+            ->setOptions([
+                'field' => 'active',
+                'primary_field' => 'id_cart_rule',
+                'route' => 'admin_cart_rule_toggle_status',
+                'route_param_name' => 'cartRuleId',
+            ])
+            )
+            ->add((new ActionColumn('actions'))
+            ->setName($this->trans('Actions', [], 'Admin.Global'))
+            ->setOptions([
+                'actions' => (new RowActionCollection())
+                    ->add((new SubmitRowAction('delete'))
+                    ->setName($this->trans('Delete', [], 'Admin.Actions'))
+                    ->setIcon('delete')
+                    ->setOptions([
+                        'method' => 'POST',
+                        'route' => 'admin_cart_rules_delete',
+                        'route_param_name' => 'cartRuleId',
+                        'route_param_field' => 'id_cart_rule',
+                        'confirm_message' => $this->trans(
+                            'Delete selected item?',
+                            [],
+                            'Admin.Notifications.Warning'
+                        ),
+                    ])
+                    ),
+            ])
+            );
+    }
```

Comment:
Should we add a modal here? cf. #17082

Reply:
yes for sure!
Those ugly alert boxes were used until 1.7.7 but it should be changed in 1.7.8. Most (if not all) the pages in develop already has the new modals.

@JevgenijVisockij, you can use `BulkDeleteActionTrait`, `DeleteActionTrait` in grid definition. You can check ManufacturerGridDefinition(or most of other grids already has it in develop branch) for example.